pilot/proxy/envoy/watcher.go

```diff
@@ -112,26 +116,28 @@ func (w *watcher) Reload() {
 
 // retrieveAZ will only run once and then exit because AZ won't change over a proxy's lifecycle
 // it has to use a reload due to limitations with envoy (az has to be passed in as a flag)
-func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration) {
-	for w.config.AvailabilityZone == "" {
+func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration, retries int) {
+	attempts := 0
+	for w.config.AvailabilityZone == "" && attempts <= retries {
 		time.Sleep(delay)
 		resp, err := http.Get(fmt.Sprintf("http://%v/v1/az/%v/%v", w.config.DiscoveryAddress, w.config.ServiceCluster, w.role.ServiceNode()))
 		if err != nil {
-			log.Infof("Error retrieving availability zone from pilot: %v", err)
+			log.Debugf("Unable to retrieve availability zone from pilot: %v", err)
```

Comment:
I think it's reasonable to keep it Info - or even Warn. It's important enough.

The problem is that the text of the message is not making clear what happens and how to fix it and what it means. 
Is pilot not reachable ? Wrong version of pilot ? Mabye this should be a fatal error, since without pilot envoy won't work - or maybe this should be handled by using some saved config/state ( when we have that)

Is pilot not aware of this machine IP and can't determine the zone ? Error should indicate that sidecar will start in fallback mode, maybe increment a varz so monitoring can detect and alert. And message should indicate workaround ( explicit setup of the zone - how ? ), and maybe link to some howto on how to fix pilot.

Reply:
more likely pod not annotated with az info

Reply:
@costinm I agree that there should be an extra log line about not enabling az aware routing if all the requests fail and then mention the explicit flag. It shouldn't be a fatal error as envoy will still work as normal just not keep traffic geographically local.

I'm not sure the reasons why it can't reach pilot so am hesitant to link to things there until I understand more about the failure cases.